Documentation/Input/CreateDataProvider.md

```diff
@@ -0,0 +1,202 @@
+
+## Namespace and folder structure
+
+> [!Important]
+> If an input system data provider is being submitted to the [Mixed Reality Toolkit repository](https://github.com/Microsoft/MixedRealityToolkit-Unity), the
+namespace **must** begin with Microsoft.MixedReality.Toolkit (ex: Microsoft.MixedReality.Toolkit.WindowsMixedReality) and the code should be
+located in a folder beneath MixedRealityToolkit.Providers (ex: MixedRealityToolkit.Providers\WindowsMixedReality).
+
+### Namespace
+
+Data providers are required to have a namespace to mitigate potential name collisions. It is recommended that the namespace includes the following components.
+
+- Company name
+- Feature area
+
+For example, an input data provider created by the Contoso company may be "Contoso.MixedReality.Input".
+
+### Recommended folder structure
+
+It is recommended that the source code for data providers be layed out in a folder heirarchy as shown in the following image.
+
+![Example folder structure](../Images/SpatialAwareness/ExampleProviderFolderStructure.png)
+
+Where ContosoInput contains the implementation of the data provider, the Editor folder contains the inspector (and any other Unity editor specific code) and Profiles
+contains one or more pre-made profiles.
+
+## Implement the data provider
+
+### Specify interface and/or base class inheritance
+
+All input system data providers must implement the [`IMixedRealityInputDeviceManager`](xref:Microsoft.MixedReality.Toolkit.Input.IMixedRealityInputDeviceManager)
+interface, which specifies the minimium functionality required by the input system. The MRTK foundation includes the [`BaseInputDeviceManager`](xref:Microsoft.MixedReality.Toolkit.Input.BaseInputDeviceManager)
+class which provides a default implementation of this required functionality. For devices that build upon Unity's UInput class, the [`UnityJoystickManager`](xref:Microsoft.MixedReality.Toolkit.Input.UnityInput.UnityJoystickManager)
+class can be used as a base class.
+
+> [!Note]
+> The `BaseInputDeviceManager` and `UnityJoystickManager` classes provide the required `IMixedRealityInputDeviceManager` implementation.
+
+``` c#
+public class WindowsMixedRealityDeviceManager : BaseInputDeviceManager, IMixedRealityCapabilityCheck
+{ }
+```
+
+> `IMixedRealityCapabilityCheck` is used by the `WindowsMixedRealityDeviceManager` to indicate that it provides support for a set of input capabilities, specifically; articulated hands,
+gaze-gesture-voice hands and motion controllers.
+
+### Implement the IMixedRealityDataProvider methods
+
+Once the class has been defined, the next step is to provide the implementation of the [`IMixedRealityDataProvider`](xref:Microsoft.MixedReality.Toolkit.IMixedRealityDataProvider)
+interface.
+
+> [!Note]
+> The `BaseInputDevicemManager` class, via the `BaseService` class, provides only an empty implementations for `IMixedRealityDataProvider` methods. The details of these methods are generally data provider specific.
```

Comment:
"provides only empty implementations" (remove an)

Reply:
fixed in next commit